### PR TITLE
feat: support nested routes

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
Hey, this is something I've been using with my website for a while now,
but I thought this could benefit others too.
Currently Vite-ssg only considers top-level routes, I've added a function that recursively traverses the route tree.
Since I've been using this myself for a while now, I'm quite confident that it does not introduce any errors but you might wanna have a closer look :)
Cheers!

### CODE REVIEW

File: src/node/build.ts [hunk 1]
```diff
@@ -75,11 +75,25 @@ export async function build(cliOptions: ViteSSGOptions = {}) {
   let indexHTML = await fs.readFile(join(out, 'index.html'), 'utf-8')
 
   const { routes } = createApp(false)
-  // ignore dynamic routes
+
+  // this function recursively extracts all paths from a given route
+  const pathsFromRoute = (prefix: string) => (route: any) => {
```
Comment (antfu) [inline src/node/build.ts:80]:
```suggestion
  const pathsFromRoute = (prefix = '') => (route: any) => {
```

**Comment by antfu** [comment]
Would you mind extracting L78-97 into a function so it would be more clear for what it's doing? Thanks.

**Comment by JonasKruckenberg** [comment]
yes, I didn't quite know how to name the function but I hope it's declarative enough 👍

File: src/node/utils.ts [hunk 2]
```diff
@@ -7,3 +7,24 @@ export function buildLog(text: string, count?: number) {
 export function getSize(str: string) {
   return `${(str.length / 1024).toFixed(2)}kb`
 }
+
+export function collectRoutePaths(routes: any[]) {
```
Comment (antfu) [inline src/node/utils.ts:11]:
```suggestion
export function routesToPaths(routes: any[]) {
```

WDYT?

Comment (antfu) [reply src/node/utils.ts:11]: Ah, and we should add the type for routes.

Comment (JonasKruckenberg) [reply src/node/utils.ts]: Yes 👍

**Comment by JonasKruckenberg** [comment]
There you go 👍 
Types are done and the function name is a lot more descriptive now thanks!

**Comment by antfu** [comment]
> exclude paths than just dynamic paths

Thought about this as well, I will add a hook to it later.

**Comment by antfu** [comment]
Both this PR and your feature request are shipped in v0.8.0

**Pull request merged**